src/main/java/sg/ncl/MainController.java

```diff
@@ -1532,6 +1536,92 @@ public String rejectJoinRequest(@PathVariable Integer teamId, @PathVariable Inte
+
+                switch (exceptionState) {
+                    case INSUFFICIENT_PERMISSION_EXCEPTION:
+                        log.warn(errorMessage + ": insufficient permission", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "You do not have permission to delete this image. Only " +
+                                                                                " team leader or creator of this image can delete the image.");
+                        break;
+                    case TEAM_NOT_FOUND_EXCEPTION:
+                        log.warn(errorMessage + ": team not found", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "Team '" + teamId + "' is not found");
+                        break;
+                    case IMAGE_NOT_FOUND_EXCEPTION:
+                        log.warn(errorMessage + ": image does not exist or not found in teams' list of images", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "The image '" + imageName + "' either does not exist or not found in your teams' list of images");
+                        break;
+                    case DETERLAB_OPERATION_FAILED_EXCEPTION:
+                        log.warn(errorMessage + ": operation failed on DeterLab", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        break;
+                    case ADAPTER_CONNECTION_EXCEPTION:
+                        log.warn(errorMessage + ": adapter connection error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        break;
+                    case ADAPTER_INTERNAL_ERROR_EXCEPTION:
+                        log.warn(errorMessage + ": adapter internal error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        break;
+                    default:
+                        log.warn(errorMessage + ": {}", imageName, teamId,  exceptionState);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        break;
+                }
+                return REDIRECT_TEAMS;
+            } else {
+                String sioMessage = new JSONObject(responseBody).getString("msg");
+
+                switch (sioMessage) {
+                    case "image still in use":
+                        log.warn(errorMessage + ": {}", imageName, teamId, sioMessage);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "Image " + "\'" + imageName + "\'" + " is still in use or busy!");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Image " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)